pkg/3scale/amp/operator/apicast_options_provider.go

```diff
@@ -195,6 +201,12 @@ func (a *ApicastOptionsProvider) productionPodTemplateLabels() map[string]string
 		labels[k] = v
 	}
 
+	if a.apimanager.Spec.Apicast.ProductionSpec.Labels != nil {
```

Comment:
no need to check.

Reply:
Addressed with this [commit ](https://github.com/3scale/3scale-operator/pull/835/commits/a92f1f214c07bead0c9650c1dbbf8003b80a09ab)